### DISCUSSION TITLE
Ficheiro de Compras: Campo References na posição errada ?

### DISCUSSION THREAD

**Discussion by Schobert** [assoft-portugal/SAF-T-AO#148] [Q&A] [answered]
Mensagem de erro ao submeter o ficheiro de Compras:
**_Field AuditFile.SourceDocuments.PurchaseInvoices.111Invoice.References it´s in the wrong position._**

A mensagem de erro indica que o campo/esturutura _References_ está na posição incorreta. A mim parece-me correto, está de acordo com o DP317/2020 e o XSD.

<Invoice><InvoiceNo>NC202054</InvoiceNo><**References**><Reference>FA2020278</Reference><InvoiceDate>2021-02-01</InvoiceDate></References><Hash>0</Hash><SourceID>DUMMY</SourceID><Period>12</Period><InvoiceDate>2020-12-01</InvoiceDate><PurchaseType>NC</PurchaseType><SupplierID>R0006</SupplierID><DocumentTotals><TaxPayable>0.00</TaxPayable><NetTotal>999999.00</NetTotal><GrossTotal>999999.00</GrossTotal></DocumentTotals></Invoice>

Será que o validador não está atualizado com a nova estrutura? ou está efetivamente algo errado?
Obrigada pela ajuda.

**Comment by cryptolopes** [comment]
Está em linha com este exemplo?

https://github.com/assoft-portugal/SAF-T-AO/blob/59f4f22bff0333192274b94406582ae026df76ec/Resources/Examples/SAF-T%20AO%20(PurchaseInvoices).xml#L101-L140

**Comment by Schobert** [comment]
Sim, está. 
![image](https://user-images.githubusercontent.com/57015106/112363544-8a375b00-8ccd-11eb-94e6-d13fa0581a12.png)

**Comment by cryptolopes** [comment, marked as answer]
> Sim, está.

Possivelmente a plataforma ainda não está a aceitar a nova estrutura.

**Comment by brunoapaulo** [comment]
Boa tarde,

Ao tentar enviar um SAF-T compras este mês deu-me este erro:
- _**Quando  AuditFile.SourceDocuments.PurchaseInvoices.Invoice.PurchaseType=NC, o campo AuditFile.SourceDocuments.PurchaseInvoices.59Invoice.References (NC C-NC+/2103CC0001) deve existir no ficheiro**_

Só deu erro a partir deste mês, já em meses anteriores temos tido notas de credito de compras e nunca deu erro.

Experimentamos colocar o xml conforme esta explicado em cima:
![Screenshot 2021-04-28 at 15 10 18](https://user-images.githubusercontent.com/43764780/116420812-3b359600-a836-11eb-99fd-d9950b27fd8c.png)

E agora dá o seguinte erro:
- _**A ordem do campo AuditFile.SourceDocuments.PurchaseInvoices.2Invoice.References no ficheiro está errada.**_

Não consigo perceber o que está errado.

Desde já obrigado pela ajuda
